app/processing/is-agreement-closed.js

```diff
@@ -0,0 +1,22 @@
+const db = require('../data')
+
+const isAgreementClosed = async (paymentRequest) => {
+  const schemeId = paymentRequest.schemeId
+  const frn = paymentRequest.frn
+  const agreementNumber = paymentRequest.agreementNumber
+  const todaysDate = new Date()
+  const agreement = await db.frnAgreementClosed.findOne({
+    where: { schemeId, frn, agreementNumber },
+    raw: true
+  })
+  if (new Date(agreement?.closureDate) < todaysDate) {
```

Comment:
closureDate is already a date so shouldn't need to parse it again.  Unless I'm missing something?

Could also just return false immediately if there's no match in the database.

Then only do the date comparison if there is?

Reply:
Suggest less than or equal to.  Just so it's clear from that day we will close, rather than after that day we will close

Reply:
originally when I was coding it I couldn't get it to work without parsing it first - but that no longer seems to be the case after swapping its type to TIMESTAMP in the DB - so removed. also changes to less than or equal to.